Sources/MapboxCoreNavigation/RouteOptions.swift

```diff
@@ -2,12 +2,17 @@ import CoreLocation
 import MapboxDirections
 
 extension RouteOptions {
-    internal var activityType: CLActivityType {
+    /**
+     The type of user activity associated with the location updates.
+     
+     SDK configures this value judging by `profileIdentifier`. You can subclass and override this property if needed.
+     */
+    @objc open var activityType: CLActivityType {
```

Comment:
I see a couple of problems with allowing for customization by making this property open:
- configuring by subclassing is not as convenient as setting a property, it's just more work,
- when overridden, this property will not persist in saving, copying (unless the SDK user provides custom support), and possibly other passes through the SDK,
- as I can see, `activityType` is not something semantically belonging to `RouteOptions` (while it's fine having it as an internal computed property, exposing it publicly should mean something),
- it adds unnecessary baggage of Obj-C runtime to a pure Swift object.

I see that this property is used only once in `NavigationService.init` and I think it could be an additional optional parameter in the initializer.

```swift
required public init(routeResponse: RouteResponse,
                     routeIndex: Int,
                     routeOptions: RouteOptions,
                     customRoutingProvider: RoutingProvider? = nil,
                     credentials: Credentials,
                     locationSource: NavigationLocationManager? = nil,
                     eventsManagerType: NavigationEventsManager.Type? = nil,
                     simulating simulationMode: SimulationMode? = nil,
                     routerType: Router.Type? = nil,
                     customAcitivityType: CLActivityType? = nil) 
```

Reply:
Subclassing is something we encourage users to do if they want to customize the behavior. Changing activity type is quite a rare use case, so IMHO it's fine to do it via subclassing rather than toggling a property.
This is a computed property, what do you mean by saving and copying here?
We tie the activity type to depend on the profile selected so from this perspective it fits `RouteOptions` purpose quite well. Why you think it should be separated?
And I didn't want to add new entities/arguments to avoid adding breaking changes.

Reply:
> Subclassing is something we encourage users to do if they want to customize the behavior. Changing activity type is quite a rare use case, so IMHO it's fine to do it via subclassing rather than toggling a property.

Do we have examples of such suggestions? If something is rare, it shouldn't mean it has to be complicated.

> This is a computed property, what do you mean by saving and copying here?

Right, this is only getter, so the storage issue is probably irrelevant.

> We tie the activity type to depend on the profile selected so from this perspective it fits RouteOptions purpose quite well. Why you think it should be separated?

`RouteOption`'s function is to store options for `Directions` or `Match` request/response. The fact that we determine `activityType` based on it, it's SDK's implementation detail that may change in the future. This makes API inflexible.

> And I didn't want to add new entities/arguments to avoid adding breaking changes.

`NavigationService` is `public` and therefore not designed for inheritance. Assuming that, would addition of an optional parameter to the function break source compatibility?

Reply:
Moved activity type to MapboxNavigationService. Please check again

Reply:
> Technically speaking, it is a source breaking change, but we frequently do it.

@S2Ler What do you think would be the case when the user gets a broken compilation with such an addition?
Also, looks like our breaking API check didn't catch it. Looks like no checks were started for this branch.

Reply:
> This was brought up multiple times, and we end up deciding that this is a rare to have a code that will break, and it is not worth it to make workarounds. So, here we are: we don't consider adding a parameter with a default value as a breaking change.

I could understand it when the class and methods are `open`. In that case, subclassing code may brake. But, when the class is just `public`, are there any breaking cases?

Reply:
Breaking API check did not run because I have issues running CI from my account... I'm trying to reenable it but meanwhile maybe you could manually trigger the workflow